### DISCUSSION TITLE
Filters and Conditions $ extension point

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#14196] [Feature Requests]
This is to track adding the ability to do custom extensions in filters and conditions with the $ prefix.

This would allow an extension author to register a $<function_name> function that could be used anywhere in filters, permissions, etc...

This function should be executed asynchronously and be provided a context that contains .env settings, directus services, etc...

This would allow for pretty much any complex condition to be integrated into filters etc... even external systems via fetch.

This would be a workaround for https://github.com/directus/directus/issues/8228 and potentially https://github.com/directus/directus/discussions/2993

**Comment by claytongulick** [comment]
Adding this to track sponsored feature request

**Comment by rijkvanzanten** [comment]
> This function should be executed asynchronously and be provided a context that contains .env settings, directus services, etc...

To expand on this: There's two main points where this needs to run: app and API. The API will have access to `.env`, the services, database, etc. The App one will only have access to the current item context, and will have to interface with the API from there

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
